src/EventService.Data/EventCommentRepository.cs

```diff
@@ -0,0 +1,98 @@
+﻿using System;
+using System.Linq;
+using System.Threading.Tasks;
+using LT.DigitalOffice.EventService.Data.Interfaces;
+using LT.DigitalOffice.EventService.Data.Provider;
+using LT.DigitalOffice.EventService.Models.Db;
+using LT.DigitalOffice.Kernel.Extensions;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.JsonPatch;
+using Microsoft.EntityFrameworkCore;
+
+namespace LT.DigitalOffice.EventService.Data;
+
+public class EventCommentRepository : IEventCommentRepository
+{
+  private readonly IDataProvider _provider;
+  private readonly IHttpContextAccessor _httpContextAccessor;
+
+  public EventCommentRepository(
+    IDataProvider provider,
+    IHttpContextAccessor httpContextAccessor)
+  {
+    _provider = provider;
+    _httpContextAccessor = httpContextAccessor;
+  }
+
+  public async Task<Guid?> CreateAsync(DbEventComment dbEventComment)
+  {
+    if (dbEventComment is null)
+    {
+      return null;
+    }
+
+    _provider.EventComments.Add(dbEventComment);
+    await _provider.SaveAsync();
+
+    return dbEventComment.Id;
+  }
+
+  public async Task<bool> EditContentAsync(Guid commentId, JsonPatchDocument<DbEventComment> request)
+  {
+    DbEventComment dbEventComment = await _provider.EventComments.FirstOrDefaultAsync(x => x.Id == commentId);
+
+    if (dbEventComment is null || request is null)
+    {
+      return false;
+    }
+
+    request.ApplyTo(dbEventComment);
+    dbEventComment.ModifiedBy = _httpContextAccessor.HttpContext.GetUserId();
+    dbEventComment.ModifiedAtUtc = DateTime.UtcNow;
+
+    await _provider.SaveAsync();
+
+    return true;
+  }
+
+  public async Task<bool> EditIsActiveAsync(Guid commentId, JsonPatchDocument<DbEventComment> request)
```

Comment:
can't is active be changed in previous method?

Reply:
I think it's better to separate from each other